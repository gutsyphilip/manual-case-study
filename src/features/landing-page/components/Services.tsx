import React from "react";

import styles from "./landing-page.module.scss";
interface Props {
  servicesList: {
    tagLine: string;
    title: string;
    description: string;
    src: string;
    srcSet: string[];
  }[];
}

const Services: React.FC<Props> = ({ servicesList }) => {
  return (
    <section className={styles.services}>
      <header className={styles.services_header}>
        <h2 className={styles.services_header_title}>What we can help with</h2>
      </header>
      <div className={styles.services_list}>
        {servicesList.map(
          ({ tagLine, title, description, src, srcSet }, index) => {
            return (
              <div
                className={styles.services_list_item}
                key={`${title.replace(" ", "+")}`}
              >
                <img
                  className={styles.services_list_item_image}
                  src={src}
                  srcSet={srcSet.join(", ")}
                  alt={title}
                />
                <article className={styles.services_list_item_caption}>
                  <h6 className={styles.services_list_item_caption_tagLine}>
                    {tagLine}
                  </h6>
                  <h3 className={styles.services_list_item_caption_title}>
                    {title}
                  </h3>
                  <p className={styles.services_list_item_caption_description}>
                    {description}
                  </p>
                </article>
              </div>
            );
          }
        )}
      </div>
    </section>
  );
};

export default Services;
