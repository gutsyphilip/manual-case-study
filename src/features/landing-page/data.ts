import HairLossImg from "../../assets/images/hair-photo.png";
import HairLossImg2x from "../../assets/images/hair-photo@2x.png";
import HairLossImg3x from "../../assets/images/hair-photo@3x.png";

import ErectileDysImg from "../../assets/images/sex-photo.png";
import ErectileDysImg2x from "../../assets/images/sex-photo@2x.png";
import ErectileDysImg3x from "../../assets/images/sex-photo@3x.png";

export const SERVICES_LIST: {
  tagLine: string;
  title: string;
  description: string;
  src: string;
  srcSet: string[];
}[] = [
  {
    tagLine: "HAIR LOSS",
    title: "Hair loss needn’t be irreversible. We can help!",
    description:
      "Hair loss needn’t be irreversible. There’s a scientifically proven way that’s most effective in keeping and regrowing your hair. It’s all to do with blocking a pesky hormone called DHT. That’s the bad guy behind hair loss. And you can keep him at bay. The choice is yours.",
    src: HairLossImg,
    srcSet: [HairLossImg2x, HairLossImg3x],
  },
  {
    tagLine: "Erectile Dysfunction",
    title: "Erections can be a tricky thing. But no need to feel down!",
    description:
      "There are plenty of reasons why you might be having difficulty in the erection department. We can help you figure out possible reasons why. And prescribe a pill if needed.",
    src: ErectileDysImg,
    srcSet: [ErectileDysImg2x, ErectileDysImg3x],
  },
];
