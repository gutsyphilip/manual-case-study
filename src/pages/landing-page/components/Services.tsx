import React from "react";

import styles from "./landing-page.module.scss";
import { SERVICES_LIST } from "../data";
interface Props {}

const Services: React.FC<Props> = (props) => {
  return (
    <section className={styles.services}>
      <header className={styles.services_header}>
        <h2 className={styles.services_header_title}>What we can help with</h2>
      </header>
      <div className={styles.services_list}>
        {SERVICES_LIST.map(
          ({ tagLine, title, description, src, srcSet }, index) => {
            return (
              <div
                className={styles.services_list_item}
                key={`${title.replace(" ", "+")}`}
              >
                <img
                  className={styles.services_list_item_image}
                  src={src}
                  srcSet={srcSet.join(", ")}
                  alt={title}
                />
                <article className={styles.services_list_item_caption}>
                  <h6 className={styles.services_list_item_caption_tagLine}>
                    {tagLine}
                  </h6>
                  <h3 className={styles.services_list_item_caption_title}>
                    {title}
                  </h3>
                  <p className={styles.services_list_item_caption_description}>
                    {description}
                  </p>
                </article>
              </div>
            );
          }
        )}
      </div>
    </section>
  );
};

export default Services;
