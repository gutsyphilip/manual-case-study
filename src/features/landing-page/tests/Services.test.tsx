import React from "react";
import { render } from "@testing-library/react";
import Services from "../components/Services";
import { SERVICES_LIST } from "../data";

const ServicesProps = {
  servicesList: SERVICES_LIST,
};
test("renders services based on data list", () => {
  const tree = render(<Services {...ServicesProps} />);
  expect(tree.asFragment()).toMatchSnapshot();
});
