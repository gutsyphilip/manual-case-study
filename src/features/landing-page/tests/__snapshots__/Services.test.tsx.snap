// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders services based on data list 1`] = `
<DocumentFragment>
  <section
    class="services"
  >
    <header
      class="services_header"
    >
      <h2
        class="services_header_title"
      >
        What we can help with
      </h2>
    </header>
    <div
      class="services_list"
    >
      <div
        class="services_list_item"
      >
        <img
          alt="Hair loss needn’t be irreversible. We can help!"
          class="services_list_item_image"
          src="hair-photo.png"
          srcset="hair-photo@2x.png, hair-photo@3x.png"
        />
        <article
          class="services_list_item_caption"
        >
          <h6
            class="services_list_item_caption_tagLine"
          >
            HAIR LOSS
          </h6>
          <h3
            class="services_list_item_caption_title"
          >
            Hair loss needn’t be irreversible. We can help!
          </h3>
          <p
            class="services_list_item_caption_description"
          >
            Hair loss needn’t be irreversible. There’s a scientifically proven way that’s most effective in keeping and regrowing your hair. It’s all to do with blocking a pesky hormone called DHT. That’s the bad guy behind hair loss. And you can keep him at bay. The choice is yours.
          </p>
        </article>
      </div>
      <div
        class="services_list_item"
      >
        <img
          alt="Erections can be a tricky thing. But no need to feel down!"
          class="services_list_item_image"
          src="sex-photo.png"
          srcset="sex-photo@2x.png, sex-photo@3x.png"
        />
        <article
          class="services_list_item_caption"
        >
          <h6
            class="services_list_item_caption_tagLine"
          >
            Erectile Dysfunction
          </h6>
          <h3
            class="services_list_item_caption_title"
          >
            Erections can be a tricky thing. But no need to feel down!
          </h3>
          <p
            class="services_list_item_caption_description"
          >
            There are plenty of reasons why you might be having difficulty in the erection department. We can help you figure out possible reasons why. And prescribe a pill if needed.
          </p>
        </article>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
